boot/assets.go

```diff
@@ -790,16 +795,39 @@ func observeSuccessfulBootAssetsForBootloader(m *Modeenv, root string, opts *boo
 		return nil, nil
 	}
 
+	byAssetName := make(map[string][]string)
```

Comment:
why the []string?  assetNames are noew unique again and have one asset assigned to them?

Reply:
I was verifying that we had only one later. But now I have rewritten it so that we verify it right away.